shared-unpackaged/is-obscured.js

```diff
@@ -26,55 +34,70 @@ export default function isObscured(element: Element): boolean {
     // would impact perf of the regular cases if we were always checking it.
-    // TODO(somewhatabstract): Consider how we might mitigate the pointer-events
-    // issue and make this call more robust.
-    const anchorRect = element.getBoundingClientRect();
-    const style =
-        ((element: any).currentStyle: ?CSSStyleDeclaration) ||
-        window.getComputedStyle(element);
-    const anchorLeft =
-        anchorRect.left +
-        parseFloat(style.marginLeft) +
-        parseFloat(style.paddingLeft) +
-        parseFloat(style.borderLeftWidth);
-    const anchorTop =
-        anchorRect.top +
-        parseFloat(style.marginTop) +
-        parseFloat(style.paddingTop) +
-        parseFloat(style.borderTopWidth);
-    const topLeftElement = document.elementFromPoint(anchorLeft, anchorTop);
-    // The bottom right corner is one less than the bounds, otherwise we
-    // can end up getting the parent of the anchor, rather than the anchor
-    // itself.
-    const anchorRight =
-        anchorRect.right -
-        parseFloat(style.marginRight) -
-        parseFloat(style.paddingRight) -
-        parseFloat(style.borderRightWidth);
-    const anchorBottom =
-        anchorRect.bottom -
-        parseFloat(style.marginBottom) -
-        parseFloat(style.paddingBottom) -
-        parseFloat(style.borderBottomWidth);
-    const bottomRightElement = document.elementFromPoint(
-        anchorRight,
-        anchorBottom,
-    );
-    // TODO(somewhatabstract): Need to cater to the case where the viewport is
-    // zoomed such that both corners are off screen but the rest isn't as in
-    // some browsers, elementFromPoint then doesn't return the element (see
-    // WB-300).
 
-    // To cope with us hitting a child of our anchor or a parent due to
-    // borders and things, we do some descendancy checks. We're ok with
-    // saying that we're visible if we hit a parent because we already
-    // checked them for visibility earlier.
-    const topLeftVisible =
-        topLeftElement &&
-        (element.contains(topLeftElement) || topLeftElement.contains(element));
-    const bottomRightVisible =
-        bottomRightElement &&
-        (element.contains(bottomRightElement) ||
-            bottomRightElement.contains(element));
-    return !topLeftVisible && !bottomRightVisible;
+    // TODO(somewhatabstract, WB-300): Need to cater to the case where the
+    // viewport is zoomed such that both corners are off screen but the rest
+    // isn't. In this case some browsers don't return the element from
+    // `elementFromPoint` then doesn't return the element.
+    // Also, consider how we might mitigate the pointer-events issue and make
+    // this call more robust.
+    const bounds = getBounds(anchorElement);
+
+    // This method does the main work, taking some coordinates and determining
+    // if our element is visible at that point or not.
+    const isVisible = (x: number, y: number): boolean => {
```

Comment:
I turned the previous code into a reusable "isVisible" call that takes coordinates. This reduced repetition and also meant we can limit the work we do...see further on...